### DISCUSSION TITLE
can't install paper type on DSM 7.2

### DISCUSSION THREAD

**Discussion by FranckDreden** [itzg/docker-minecraft-server#2689] [Q&A]
Hi, 
I'm trying to follow this tutorial (/watch?v=0V1c33rqLwA) on YT, to install ITZG's container on a DS1520+ with DSM 7.
Container manager is up to date (20.10.23-1437), as is DSM (7.2.1-69057 U4).

Here is the YAML conf:
version: "3.8"

services:
  mc:
    image: itzg/minecraft-server
    tty: true
    stdin_open: true
    ports:
      - "25565:25565"
    environment:
      EULA: "TRUE"
      TYPE: "PAPER"
    volumes:
      # attach the relative directory 'data' to the container's /data path
      - ./data:/data
      
And here are the logs when I execute the container project:

minecraft-java-mc-1
date,stream,content
2024/03/02 08:52:05,stdout,[init] ERROR: failed to download paper
"2024/03/02 08:52:05,stdout,	at java.base/java.lang.Thread.run(Unknown Source)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)"
"2024/03/02 08:52:05,stdout,	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)"
"2024/03/02 08:52:05,stdout,	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)"
"2024/03/02 08:52:05,stdout,	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:613)"
2024/03/02 08:52:05,stdout,Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
"2024/03/02 08:52:05,stdout,		... 13 common frames omitted"
"2024/03/02 08:52:05,stdout,		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:103)"
"2024/03/02 08:52:05,stdout,	Suppressed: java.lang.Exception: #block terminated with an error"
"2024/03/02 08:52:05,stdout,	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:159)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine.execute(CommandLine.java:2170)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine.access$1500(CommandLine.java:148)"
"2024/03/02 08:52:05,stdout,	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)"
"2024/03/02 08:52:05,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:35)"
"2024/03/02 08:52:05,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:109)"
"2024/03/02 08:52:05,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.useCoordinates(InstallPaperCommand.java:152)"
"2024/03/02 08:52:05,stdout,	at reactor.core.publisher.Mono.block(Mono.java:1712)"
"2024/03/02 08:52:05,stdout,	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:101)"
"2024/03/02 08:52:05,stdout,	at reactor.core.Exceptions.propagate(Exceptions.java:408)"
2024/03/02 08:52:05,stdout,reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
2024/03/02 08:52:05,stdout,[mc-image-helper] 07:52:05.181 ERROR : 'install-paper' command failed. Version is 1.38.5
2024/03/02 08:50:32,stdout,[init] Resolving type given PAPER
2024/03/02 08:50:32,stdout,[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 1 1000 1000 70 Mar  2 07:29 /data'
2024/03/02 08:35:30,stdout,[init] ERROR: failed to download paper
"2024/03/02 08:35:29,stdout,	at java.base/java.lang.Thread.run(Unknown Source)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)"
"2024/03/02 08:35:29,stdout,	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)"
"2024/03/02 08:35:29,stdout,	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)"
"2024/03/02 08:35:29,stdout,	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:613)"
2024/03/02 08:35:29,stdout,Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
"2024/03/02 08:35:29,stdout,		... 13 common frames omitted"
"2024/03/02 08:35:29,stdout,		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:103)"
"2024/03/02 08:35:29,stdout,	Suppressed: java.lang.Exception: #block terminated with an error"
"2024/03/02 08:35:29,stdout,	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:159)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine.execute(CommandLine.java:2170)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine.access$1500(CommandLine.java:148)"
"2024/03/02 08:35:29,stdout,	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)"
"2024/03/02 08:35:29,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:35)"
"2024/03/02 08:35:29,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:109)"
"2024/03/02 08:35:29,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.useCoordinates(InstallPaperCommand.java:152)"
"2024/03/02 08:35:29,stdout,	at reactor.core.publisher.Mono.block(Mono.java:1712)"
"2024/03/02 08:35:29,stdout,	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:101)"
"2024/03/02 08:35:29,stdout,	at reactor.core.Exceptions.propagate(Exceptions.java:408)"
2024/03/02 08:35:29,stdout,reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
2024/03/02 08:35:29,stdout,fo  docker pull [mc-image-helper] 07:35:29.839 ERROR : 'install-paper' command failed. Version is 1.38.5
2024/03/02 08:33:57,stdout,[init] Resolving type given PAPER
2024/03/02 08:33:57,stdout,[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 1 1000 1000 70 Mar  2 07:29 /data'
2024/03/02 08:30:51,stdout,[init] ERROR: failed to download paper
"2024/03/02 08:30:51,stdout,	at java.base/java.lang.Thread.run(Unknown Source)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)"
"2024/03/02 08:30:51,stdout,	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153)"
"2024/03/02 08:30:51,stdout,	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)"
"2024/03/02 08:30:51,stdout,	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:613)"
2024/03/02 08:30:51,stdout,Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
"2024/03/02 08:30:51,stdout,		... 13 common frames omitted"
"2024/03/02 08:30:51,stdout,		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:103)"
"2024/03/02 08:30:51,stdout,	Suppressed: java.lang.Exception: #block terminated with an error"
"2024/03/02 08:30:51,stdout,	at me.itzg.helpers.McImageHelper.main(McImageHelper.java:159)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine.execute(CommandLine.java:2170)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine.access$1500(CommandLine.java:148)"
"2024/03/02 08:30:51,stdout,	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)"
"2024/03/02 08:30:51,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:35)"
"2024/03/02 08:30:51,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.call(InstallPaperCommand.java:109)"
"2024/03/02 08:30:51,stdout,	at me.itzg.helpers.paper.InstallPaperCommand.useCoordinates(InstallPaperCommand.java:152)"
"2024/03/02 08:30:51,stdout,	at reactor.core.publisher.Mono.block(Mono.java:1712)"
"2024/03/02 08:30:51,stdout,	at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:101)"
"2024/03/02 08:30:51,stdout,	at reactor.core.Exceptions.propagate(Exceptions.java:408)"
2024/03/02 08:30:51,stdout,reactor.core.Exceptions$ReactiveException: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
2024/03/02 08:30:51,stdout,[mc-image-helper] 07:30:51.268 ERROR : 'install-paper' command failed. Version is 1.38.5
2024/03/02 08:29:16,stdout,[init] Resolving type given PAPER
2024/03/02 08:29:16,stdout,[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 1 1000 1000 0 Mar  2 07:28 /data'
2024/03/02 08:29:16,stdout,[init] Changing ownership of /data to 1000 ...

I'm completely new to Container Manager and I don't know what to do to follow this tutorial. All I'd like is set up this Minecraft Server at home for my kids to play.

Thanks for any and all help anyone could give me.

**Comment by itzg** [comment]
Looking at the root cause in your logs

```
Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: api.papermc.io/172.67.72.198:443
```

You seem to have a networking issue. It might be just temporary but if not it might be either in Container Manager's setup or your LAN.

**Reply by FranckDreden** [reply]
I have several packages installed on this NAS, everything working over WAN and LAN perfectly fine. I have Teamspeak and Tautulli in Container Manager and Plex as a package directly installed. So I doubt this is a LAN setup problem. Although I did see this timeout, reason why I tried multiple times.

I thought this might be a problem with firewalling and added an exclusion rule to this container projet. To no avail.

Hence my question here. Is there a mirror I can try to configure the project to go look for the download, other than api.papermc.io?

**Comment by FranckDreden** [comment]
well, apparently creating a firewall rule for "Docker minecraft" wasn't enough, I had to create a exclusion rule for the Container Manager bridge network inside which my docker project runs.

Now I face another problem, which I cannot fathom.
I want to migrate our previous world (where my kids played for hours), which is a vanilla world, into this paper/spigot one, I'm creating on Docker/Synology NAS.
I copied the content of our world into the separate folders: world / nether / the end, respectively of their nature. But now when I restart the server, I've got these errors: 

minecraft-java-mc-1
date,stream,content
"2024/03/03 14:38:05,stdout,[m2024-03-03T13:38:05.750Z	INFO	mc-server-runner	Done"
"2024/03/03 14:38:00,stdout,	at java.lang.Thread.run(Unknown Source) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at io.papermc.paperclip.Paperclip.lambda$main$0(Paperclip.java:42) ~[app:?]"
"2024/03/03 14:38:00,stdout,	at org.bukkit.craftbukkit.Main.main(Main.java:326) ~[paper-1.20.4.jar:git-Paper-436]"
"2024/03/03 14:38:00,stdout,	at net.minecraft.server.Main.main(Main.java:205) ~[paper-1.20.4.jar:git-Paper-436]"
"2024/03/03 14:38:00,stdout,	at net.minecraft.world.level.storage.LevelStorageSource.validateAndCreateAccess(LevelStorageSource.java:439) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at net.minecraft.world.level.storage.LevelStorageSource$LevelStorageAccess.<init>(LevelStorageSource.java:533) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at net.minecraft.util.DirectoryLock.create(DirectoryLock.java:23) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at java.nio.channels.FileChannel.open(Unknown Source) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at java.nio.channels.FileChannel.open(Unknown Source) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at sun.nio.fs.UnixFileSystemProvider.newFileChannel(Unknown Source) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[?:?]"
"2024/03/03 14:38:00,stdout,	at sun.nio.fs.UnixException.translateToIOException(Unknown Source) ~[?:?]"
2024/03/03 14:38:00,stdout,java.nio.file.AccessDeniedException: ./world/session.lock
2024/03/03 14:38:00,stdout,[31	1m[13:38:00 ERROR]: Failed to start the minecraft server
2024/03/03 14:38:00,stdout,"[13:38:00 INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, name=PROD]

Apparently caused by an "accessDeniedException" for some files, alas, the permission seems to correctly be set (Administrators and Admin group, including root, is set to Total Control. Users are read only).

I'm realling blocked by this problem, I can't seem to find a solution.

If you please are willing to help me.

**Reply by itzg** [reply]
> Now I face another problem, which I cannot fathom.
> I want to migrate our previous world (where my kids played for hours), which is a vanilla world, into this paper/spigot one

It's not documented, but it just so happens the WORLD copy feature will perform that conversion for you

https://docker-minecraft-server.readthedocs.io/en/latest/misc/world-data/

**Reply by LilVapers** [reply]
Hello Franck, I am having the same session.lock issue when trying to use my old world. I am using fabric type but its the exact same issue. I posted under the Q&A about it yesterday. Did you figure out a solution? I wasn't sure which variables to use to copy a world on Synology docker.

**Reply by FranckDreden** [reply]
Sorry to reply this late, actually I gave up, since I wasn't successfull. It seems, as I've understood it, that the session.lock is server related and created for the first time on server generation.
BUT, world can be migrated over servers (even between multi and single player), without this session.lock, thus it'll be recreated. What happens when you do that is players start this world with no recipes, no level, no nothing. Just as a new world generated.
So, without this file, supposedly, you could start the game anew with the given map (world, nether and end) and be gone with it.
Alas, it didn't work, whatever I tried, nothing worked.
Plus, you add two factors:
Syno and Docker on syno are quite closed, not just normal open Linuxes where you can do whatever you want and fiddle the way you'd like to (without taking risks on your system, that is)
And secondly, NAS's aren't quite strong enough to maintain a minecraft server.

I firstly created a Java MC server directly as a package, but server crashed with the first redstone farms and automation farms. I then tried creating this server on Docker, in order to have a better grip on the server than in Syno's packages, and fiddle around better. Alas, couldn't really migrate my world over.

My capacities are limited, my comprehension on this subject too.

All in all, I decided to move on to Realms, which is not that expensive, and sufficient to let my kids and I play. Migrated my world perfectly fine.

Sorry buddy, I wish I could've helped you better, but alas, I couldn't help myself! :(

**Reply by LilVapers** [reply]
Hey, thanks for getting back to me anyways. I've used this itzg minecraft docker image for every server and this is my first time experiencing this. 

Not too long ago, I played through a custom map named "Diversity 3" and to play that I had to use an already existing world. I believe all I had to do was keep the world as a zip file and then set a variable WORLD = Diversity3.zip. Now when I try WORLD = or LEVEL =, it just sets the name of the level to the zip file name. And like you said, DSM is hard to work with so I feel like I have tried everything. 

If I figure anything out, I will make sure to post it here.

**Reply by LilVapers** [reply]
Hello Franck, good news! I figured out how to use my old world. So basically you make a fresh container and let it generate a world. Then you delete that world file and replace it with your old one. Here is where the "java.nio.file.AccessDeniedException" error will begin. Basically what happens is when you make the container, a user named "1000" is applied to all of the applicable folders. But when you copy your old world over, the permissions user will likely be "Owner" or "Your Name" not "1000" so thats why there was no access for your session.lock file. Because it is the same case for files, they need permission for the user "1000". So another issue is I can't figure out how to manually assign the user "1000" to whatever file or folder I choose. They are not a valid user, it doesn't even work to create a user named "1000" on DSM and assign them that way. The user is assigned when the container is initially created. So what I did was I added a volume setting for /world that was read only so it would have access to this folder on startup. This worked for me. Unfortunately, this same solution does not work for mods. I remember for mods, all I used to do was drag and drop the mod file in the mods folder. Now I have this "java.nio.file.AccessDeniedException" issue for every mod and datapack I am trying to add. If I figure that out, I will update here. Also, if anyone has more knowledge of linux systems and wants to get in a call with me, we could probably put together a decent guide for DSM systems. 


![Screenshot 2024-03-13 165705](https://github.com/itzg/docker-minecraft-server/assets/88253560/a699c98f-fada-4b05-8d4e-f08627990715)

To see this, just right click a folder or file and check its properties.

![Screenshot 2024-03-13 151425](https://github.com/itzg/docker-minecraft-server/assets/88253560/69f71a70-88e9-4ff6-83d6-4bf80b4183bf)
